src/node_options.cc

```diff
@@ -0,0 +1,310 @@
+  { "v8_options", no_argument, nullptr, OPT_V8_OPTIONS,
+    "print v8 command line options" },
+  { "abort-on-uncaught-exception", no_argument, nullptr, OPT_ABORT_UNCAUGHT,
+    "abort on uncaught exception" },
+  { "abort_on_uncaught_exception", no_argument, nullptr, OPT_ABORT_UNCAUGHT,
+    "abort on uncaught exception" },
+  { "expose-internals", no_argument, nullptr, OPT_EXPOSE_INTERNALS,
+    "expose internal modules" },
+  { "expose_internals", no_argument, nullptr, OPT_EXPOSE_INTERNALS,
+    "expose internal modules" },
+  { "debug", optional_argument, nullptr, OPT_DEBUG, "enable debug mode" },
+  { "debug-brk", optional_argument, nullptr, OPT_DEBUG_BRK,
+    "break before starting" },
+  { "debug-port", required_argument, nullptr, OPT_DEBUG_PORT,
+    "specify debug port (defaults to 5858)" },
+#if defined(NODE_HAVE_I18N_SUPPORT)
+  { "icu-data-dir", required_argument, nullptr, OPT_ICU_DATA_DIR },
+#endif
+  { nullptr, 0, nullptr, 0, "" }
+};
+
+NodeOptions::NodeOptions() {
+}
+
+NodeOptions::~NodeOptions() {
+}
+
+void NodeOptions::PrintHelp() {
+  printf("Usage: iojs [options] [ -e script | script.js ] [arguments] \n"
+         "       iojs debug script.js [arguments] \n"
+         "\n"
+         "Options:\n");
+  for (size_t i = 0; i < ARRAY_SIZE(longopts); i++) {
+    if (longopts[i].name == nullptr)
+      continue;
+    if (longopts[i].val < 1000) {
+      printf("\t-%c, --%-30s %-50s\n",
+             longopts[i].val,
+             longopts[i].name,
+             longopts[i].desc);
+    } else {
+      printf("\t    --%-30s %-50s\n", longopts[i].name, longopts[i].desc);
+    }
+  }
+
+  printf("\n");
+
+  printf("Environment variables:\n"
+#ifdef _WIN32
+         "\t  NODE_PATH              ';'-separated list of directories\n"
+#else
+         "\t  NODE_PATH              ':'-separated list of directories\n"
+#endif
+         "\t                          prefixed to the module search path.\n"
+         "\t  NODE_DISABLE_COLORS     Set to 1 to disable colors in the REPL\n"
+#if defined(NODE_HAVE_I18N_SUPPORT)
+         "\t  NODE_ICU_DATA           Data path for ICU (Intl object) data\n"
+#if !defined(NODE_HAVE_SMALL_ICU)
+         "\t                          (will extend linked-in data)\n"
+#endif
+#endif
+         "\n"
+         "Documentation can be found at https://iojs.org/\n");
+}
+
+void NodeOptions::ParseArgs(int* argc,
+                            const char** argv,
+                            int* exec_argc,
+                            const char*** exec_argv,
+                            int* v8_argc,
+                            const char*** v8_argv) {
+  const unsigned int nargs = static_cast<unsigned int>(*argc);
+  const char** new_exec_argv = new const char*[nargs];
+  const char** new_v8_argv = new const char*[nargs];
+  const char** new_argv = new const char*[nargs];
+  const char** local_preload_modules = new const char*[nargs];
+
+  char** largv = const_cast<char**>(argv);
```

Comment:
Can you add a comment here explaining we're mutating the strings vector but not the strings themselves?